vitrivr-engine-module-m3d/src/main/kotlin/org/vitrivr/engine/model3d/renderer/ExternalRenderer.kt

```diff
@@ -47,21 +32,101 @@ class ExternalRenderer : Closeable {
      * @param windowOptions The [WindowOptions] to use for rendering.
      * @param renderOptions The [RenderOptions] to use for rendering.
      */
+    @Synchronized
     fun render(model: Model3d, cameraPositions: List<Vector3f>, windowOptions: WindowOptions, renderOptions: RenderOptions): List<BufferedImage> {
-        /* Create request and write it to stream. */
+        check(!this.closed) { "ExternalRenderer is closed and cannot be used for processing." }
+        var process = this.process
+        if (process == null || !process.isAlive()) {
+            process?.close()
+            process = RenderProcess()
+            this.process = process
+        }
+
+        /* Send request. */
         val request = RenderRequest(model, cameraPositions, windowOptions, renderOptions)
-        this.oos.writeObject(request)
-        this.oos.flush()
+        val response = process.send(request)
 
-        /* Read request. */
-        val image = this.ois.readObject() as? RenderResponse ?: throw IllegalStateException("Could not parse model.")
-        return image.images()
+        /* Return images . */
+        return response.images()
     }
 
     /**
      * Closes the [ExternalRenderer] and the associated process.
      */
     override fun close() {
-        this.process.destroy()
+        if (!this.closed) {
+            this.closed = true
+            this.process?.close()
+            this.process = null
+        }
+    }
+
+    /**
+     * A [RenderProcess] is a helper class that wraps the [Process] used by the [ExternalRenderer].
+     */
+    private inner class RenderProcess: Closeable {
+
+        /** The [Process] used by the [ExternalRenderer]. */
+        val process: Process
+
+        /** The [ObjectOutputStream] used by the [ExternalRenderer]. */
+        val oos: ObjectOutputStream
+
+        /** The [ObjectInputStream] used by the [ExternalRenderer]. */
+        val ois: ObjectInputStream
+
+        init {
+            val javaHome = System.getProperty("java.home")
+            val javaBin = "$javaHome/bin/java"
```

Comment:
Since you are always calling this from a JVM, you can also just look up where the current JVM is running using `ProcessHandle.current().info().command()`. This might be desirable in case there are multiple JVMs present on a given system and the used one is not the default one.